### PR TITLE
[ZEPPELIN-5425] Polish Interpreter class

### PR DESCRIPTION
**Pull request by cuspymd**
### What is this PR for?
- Move `interpolate()` function into `AbstractInterpreter` class which is only one caller of this function
- Fix IDE warning messages

### What type of PR is it?
[Refactoring]

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-5425

### How should this be tested?
* Unit tests and CI

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by Reamer** [comment]
I restarted the CI because many tests failed, which does not seem to be related to these changes.

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/AbstractInterpreter.java [hunk 1]
```diff
@@ -47,6 +51,35 @@ public InterpreterResult interpret(String st,
     return internalInterpret(st, context);
   }
 
+  static String interpolate(String cmd, ResourcePool resourcePool) {
+    Pattern zVariablePattern = Pattern.compile("([^{}]*)([{]+[^{}]*[}]+)(.*)", Pattern.DOTALL);
```
Comment (zjffdu) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/AbstractInterpreter.java:55]: Move it outside as static variable ?

Comment (Reamer) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/AbstractInterpreter.java:55]: With `static` and `final` :grin: 
https://rules.sonarsource.com/java/tag/performance/RSPEC-4248

Comment (cuspymd) [reply zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/AbstractInterpreter.java]: I fixed. But No difference was found from the execution time of `ZeppCtxtVariableTest`.
Below is the execution time of 18 tests of `ZeppCtxtVariableTest`. (measured 10 times each)
statistics | before (ms) | after (ms) 
-------|--------|-------
average | 235.6| 234.6
std. dev. | 12.6 | 26.5

**Comment by zjffdu** [comment]
LGTM

**Pull request closed without merging**